src/debugger/ReferenceMap.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+// A map with an incrementing counter as the keys
+// Used to store references to variable collections since DebugProtocol
+// specifies fetching variable collections via unique IDs
+export class ReferenceMap<T> {
+  constructor(start: number) {
+    this._start = start;
+    this._counter = start;
+    this._mapping = new Map();
```

Comment:
You can probably use an Array here if you're always mapping int => value, they work efficiently with holes as long as you don't access the holes.